src/Kernel/Process.class.st

```diff
@@ -604,6 +604,32 @@ Process >> signalException: anException [
 	oldList ifNotNil: [self resume]
 ]
 
+{ #category : 'testing' }
+Process >> state [
+
+	self == Processor activeProcess ifTrue: [ ^ #active ].
+	"A process is #terminated if one of the following conditions is met:
+	(1) the receiver is a defunct process (suspendedContext = nil or pc = nil)
+	(2) the receiver is suspended in the endProcess method (we cannot use pragmas...
+	pragmas require the compiler, it generates an evil dependency.
+	It also is not available in minimal images)"
+	(suspendedContext isNil or: [
+		 suspendedContext isDead or: [
+			 suspendedContext method == (self class >> #endProcess) ] ])
+		ifTrue: [ ^ #terminated ].
+	"Any other process without a suspendingList, or which cannot
+	possibly be present in that list because it is empty,
+	is considered #suspended. This is either a new process that has
+	never been sent #resume, or one that has been sent #suspend."
+	(myList isNil or: [ myList isEmpty ]) ifTrue: [ ^ #suspended ].
```

Comment:
Hi, I think here's an opportunity for further simplification: 
`myList ifNil: [ ^ #suspended ].`
should suffice here, because the other part of the condition (`or: [ myList isEmpty ]`) should no longer be necessary. It used to be relevant for older VMs but unfortunately, I can't find the discussion where this was explained (will post when come across it again). So just FYI :)

Reply:
Oh nice! Yeah, I wondered a little about that—notice I updated the comment, thinking "hmm, we check if the list is empty, but what about if there's another process in that list yet the receiver is not? But that clearly shouldn't happen in a well-designed VM, so why the empty check?" Maybe should have said something, except, it seems like PRs get a lot more engagement than comments on old issues, and I didn't want to delay _making_ the PR anyway. Glad you chimed in!

Reply:
Hi @daniels220 , FYI I've found the discussion regarding the `myList isEmpty` check (see the third post from the top by Vanessa Freudenberg): https://lists.squeakfoundation.org/archives/list/vm-dev@lists.squeakfoundation.org/thread/CGCZH3WM742AGJ7RUGFHGJAHCRGL77ZL/